File: frontend/src/api/stocks.ts
import axios from "axios"

const BASE = "http://localhost:8000"

export interface Stock {
  symbol:     string
  name:       string
  price:      number
  change:     number
  change_pct: number
  volume:     string
  high:       number
  low:        number
}

export interface ExchangeData {
  exchange: string
  stocks:   Stock[]
  summary:  string
  as_of:    string
}

export interface StocksResponse {
  nyse: ExchangeData
  bse:  ExchangeData
}

export const stocksApi = {
  get:     () => axios.get<StocksResponse>(`${BASE}/stocks`),
  refresh: () => axios.get<StocksResponse>(`${BASE}/stocks?force=true`),
}

File: backend/models/stocks.py
from pydantic import BaseModel

class Stock(BaseModel):
    symbol:       str
    name:         str
    price:        float
    change:       float
    change_pct:   float
    volume:       str
    high:         float
    low:          float

class ExchangeData(BaseModel):
    exchange:     str
    stocks:       list[Stock]
    summary:      str
    as_of:        str